Migration from tb68 to tb78

itagagaki opened the pull request:
Based on the TB68 version, I modified it to work on TB78. Please review!

itagagaki commented on content/folderPane.js:
```diff
@@ -180,6 +180,6 @@
   };
   
   /* Refresh pane */
-  mainWindow.gFolderTreeView.mode = mainWindow.gFolderTreeView.mode;
```
This line was added by the commit '[Fix folder sort on startup](https://github.com/protz/Manually-Sort-Folders/commit/68314ddf4556244e54c5ed13e2b3db5b402c5d0f)'
But this is causing the exception on restarting TB78.
See if this is necessary in TB78 as well. If so, we will need to find out alternative solution.

msprotz commented:
Wonderful! I'll give this a go shortly. Can you fix the merge conflicts in the meanwhile?

itagagaki commented:
I have fixed two merge conflicts.
One of them is the version number, which should eventually be updated at the time of release.
The other is a Makefile that I have modified for myself. I put it back in.

protz commented:
Thanks! This is great -- I've ported all of your commits to master but I also cleaned up the history because there was some unrelated development work there -- I moved it to a branch. Another thing I did was I bumped the minimum version to 78, because I tried your patch with 68 and I couldn't find the entry in the tools menu, or the addons preference page.

Questions for you:
- can you confirm that I didn't miss any change from your branch? 
- does this add an entry into the hamburger menu? I know that the UI changes in Thunderbird have confused a lot of people, and with newer versions users tend to have a hard time finding the menu entry

I'm going to submit this version up for review at Thunderbird addons -- I'll be happy to issue a new dot-release if need be.

Thanks very much again!

itagagaki commented:
Yes, I would like to find time to check the master.
I'm pretty sure the hamburger menu doesn't have an entry registered on it. I didn't notice that. I'll find time to look into this too to see if there's a way to deal with it. I'm sure the TB78 version will be better released after we investigate this issue.

itagagaki commented:
I confirmed that you didn't miss any change from my branch. Perfect.

I've also successfully added an entry to the hamburger menu, and it seems to work fine, except for an issue with the label being displayed a bit to the left of the other items. Please check it out. And if it's OK, merge it into master.

Also, I think you made the right decision to change the minimum version to 78. But just wait. I'd like to try to get it to work on 68 as well. I don't know if I can do it.

lgbaldoni commented:
> Also, I think you made the right decision to change the minimum version to 78. But just wait. I'd like to try to get it to work on 68 as well. I don't know if I can do it.

Looks like wasted effort as 68.12 is the last release from the old ESR branch.

itagagaki commented:
> Looks like wasted effort as 68.12 is the last release from the old ESR branch.

Maybe so. But I've done it :-)

So please port content/scripts/messenger.js, background.js and manifest.json

lgbaldoni commented:
> Maybe so. But I've done it :-)

Kudos to you:)

protz commented:
Hi @itagagaki 

This great and I was able to confirm this works. the reason the button doesn't look right is because of a typo: `subview-button` instead of `subviewbutton`. I found this by using the DOM inspector and comparing your button with the one right underneath and trying to see what the issue was.

I'll release a new version with this fix.

Thanks again for your contributions!

If you love this addon, we should chat by email about how to make sure it's compatible with Thunderbird versions past 78. Let me know.

itagagaki commented:
@protz 

Oh, the subviewbutton was an easy mistake. Thank you for the investigation.

I'm happy to hear that 2.0.4 has been released.

This add-on is a must for me as well. I'll continue to help in the future, albeit in a small way.
You know my email address so we can chat anytime.
See you soon!
